gistUpload.js

```diff
@@ -14,17 +15,40 @@ function uploadFile(contents, gist) {
     { files: post },
   )
     .then((response) => {
-      console.log(response.data.html_url);
+      console.log(`Your gist is done uploading.
+View it at: ${response.data.html_url}`);
       return (response.data.html_url);
     })
     .catch(error => (error.response.status));
 }
 
+function addToFile(chunk) {
+  finalString += chunk;
+}
+
 function readFile(fileToContent, gistToName) {
-  fs.readFile(fileToContent, (error, contents) => (
-    error ? console.log(error) : uploadFile(contents.toString(), gistToName)
-  ));
+  let bytesRead = 0;
+  const totalSize = fileSystem.statSync(process.argv[2]).size;
+  console.log(`Total size: ${totalSize}`);
+  const readStream = fileSystem.createReadStream(fileToContent);
+
+  readStream.on('data', (chunk) => {
+    try {
+      addToFile(chunk);
+      bytesRead = chunk.length + bytesRead;
+      console.log(`${chunk.length} bytes in this chunk.\n`);
+      const prog = (Math.round((readStream.bytesRead / totalSize) * 100) - 15);
+      console.log(`${prog}% of file has been read`);
+      console.log(`${bytesRead} bytes read so far\n`);
+    } catch (error) {
+      console.log(error);
+    }
+  });
+
+  readStream.on('close', () => {
+    uploadFile(finalString, gistToName);
+  });
 }
 
 
-module.exports = { uploadFile, readFile };
+module.exports = { readFile, uploadFile };
```

Comment:
Do you need to export `uploadFile` ?

Reply:
Yes, because I call it in my gistUpload.test.js